lib/buffer.js

```diff
@@ -326,13 +326,34 @@ Object.defineProperty(Buffer.prototype, 'offset', {
 function slowToString(encoding, start, end) {
   var loweredCase = false;
 
-  start = start >>> 0;
-  end = end === undefined || end === Infinity ? this.length : end >>> 0;
+  if (start >= this.length || end <= 0 || Number.isNaN(end))
+    return '';
+
+  /*
+   * if unsigned 32 bit value (>>> 0) of `start` is not the same as `start`,
+   * use `0` as the default value. Similarly, if the unsigned 32 bit value of
+   * `end` is not the same as `end`, then use the length of the buffer as the
+   * end value.
+   *
+   * Note that, the comparison is done with abstract equality operator (!=) to
+   * allow stringified numbers ('1') and Number objects (new Number(5)).
+   *
+   * `+` is required so that octal/hex/binary strings are handled correctly.
+   * `parseInt` is required so upper range integers (2^53) don't wrap
+   */
+  if (start < 0 || start >>> 0 !== parseInt(+start))
```

Comment:
Why is `parseInt()` used here and below if the value is already cast to an integer with the `+` operator?